File: PruebaPicasso/app/src/main/java/es/iesnervion/avazquez/pruebapicasso/fragments/ImageListFragment.java
package es.iesnervion.avazquez.pruebapicasso.fragments;

import android.os.Bundle;
import android.provider.ContactsContract;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.ListView;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;
import androidx.fragment.app.Fragment;
import androidx.lifecycle.ViewModelProviders;

import java.util.ArrayList;
import java.util.Arrays;

import es.iesnervion.avazquez.pruebapicasso.R;
import es.iesnervion.avazquez.pruebapicasso.adapter.ImageAdapter;
import es.iesnervion.avazquez.pruebapicasso.viewModel.SharedViewModel;

public class ImageListFragment extends Fragment {

    SharedViewModel viewModel;
    ListView listView;


    public ImageListFragment() {
    }

    @Nullable
    @Override
    public View onCreateView(@NonNull LayoutInflater inflater, @Nullable ViewGroup container, @Nullable Bundle savedInstanceState) {
        View view = inflater.inflate(R.layout.image_list, container, false);
        viewModel = ViewModelProviders.of(getActivity()).get(SharedViewModel.class);

        listView = view.findViewById(R.id.listViewImagenes);

        ArrayList<String> listaURLs = viewModel.getListaURLs().getValue();
        if(listaURLs == null){
            listaURLs = new ArrayList<>();
            listaURLs.add("https://love2dev.com/img/error-message-laptop-1920x1297.jpg");
        }

        ImageAdapter adapter = new ImageAdapter(listaURLs, getActivity().getBaseContext());
        listView.setAdapter(adapter);
        return view;
    }
}

File: EjemploKotlinRetrofit2/app/src/main/java/es/iesnervion/avazquez/ejemplokotlinretrofit2/fragments/ImageListFragment.kt
package es.iesnervion.avazquez.ejemplokotlinretrofit2.fragments


import android.os.Bundle
import androidx.fragment.app.Fragment
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import androidx.lifecycle.ViewModelProviders
import androidx.recyclerview.widget.RecyclerView

import es.iesnervion.avazquez.ejemplokotlinretrofit2.R
import es.iesnervion.avazquez.ejemplokotlinretrofit2.adapter.ImageAdapter
import es.iesnervion.avazquez.ejemplokotlinretrofit2.viewModel.SharedViewModel

/**
 * A simple [Fragment] subclass.
 */
class ImageListFragment : Fragment() {

    lateinit var viewModel : SharedViewModel
    lateinit var recyclerViewImagenes : RecyclerView
    lateinit var listadoURLs : ArrayList<String>
    override fun onCreateView(
        inflater: LayoutInflater, container: ViewGroup?,
        savedInstanceState: Bundle?
    ): View? {

            // Inflate the layout for this fragment
            val view = inflater.inflate(R.layout.fragment_image_list, container, false)
            viewModel = ViewModelProviders.of(activity!!).get(SharedViewModel::class.java)

            recyclerViewImagenes = view!!.findViewById(R.id.recyclerViewImagenes)
            //recyclerViewImagenes.adapter = RecyclerAdapter(viewModel.getURLsFotos(), activity!!)
        recyclerViewImagenes.adapter = ImageAdapter()

/*

        // Create the observer which updates the UI.
        val razaElegidaObserver = Observer<ArrayList<String>> { observer ->
            // Update the UI
            //recyclerViewImagenes.adapter = RecyclerAdapter(viewModel.getURLsFotos(), activity!!)
            recyclerViewImagenes.adapter = ImageAdapter()
        }

        // Observe the LiveData, passing in this activity as the LifecycleOwner and the observer.
        viewModel.getMutableLiveDataFotos().observe(this, razaElegidaObserver)

*/

        return view
    }


}
